backend/src/routes/spotify.js

```diff
@@ -0,0 +1,156 @@
+const express = require('express');
+const SpotifyWebApi = require('spotify-web-api-node');
+const crypto = require('crypto');
+const { firebase } = require('../firebase-init.js');
+const { authMiddleware } = require('./auth');
+
+const db = firebase.firestore();
+require('dotenv').config();
+
+const Spotify = new SpotifyWebApi({
+  clientId: process.env.SPOTIFY_CLIENT_ID,
+  clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
+  redirectUri: 'https://hopscotch-backend.herokuapp.com/api/spotify/oauth-callback',
```

Comment:
Should we put all instances of internal URLs into an env file? So when we deploy we use an env file w say
```
BASE_URL=https://hopscotch-backend.herokuapp.com
```
and for testing we use
```
BASE_URL=http://localhost:5000
```
and then
```
redirectUri: `${process.env.BASE_URL}/api/spotify/oauth-callback`
```
non-blocking, but let's do this on a separate PR

Edit: realized you put redirectHost as an optional param in `handleLink`. This could work, but let's not have the user decide where to redirect to.